### PR TITLE
Add branching information to contribution guide

### PR DESCRIPTION
**Pull request by guw**
@vazexqi we should discuss whether we should abandon the `release` branch completely and go directly from develop into master.

### CODE REVIEW

**Comment by guw** [comment]
@forcedotcom/ide FYI

**Comment by vazexqi** [comment]
@guw 

1. We should discuss with Jason about the release model. I do _think_ that with our current release cycles that we can keep this simpler until we GA. My current scripts/publish.js sort of assumes develop -> master directly without the release branch. We can validate the feasibility of this model when we do our first publish to the VS Code Marketplace.
1. Do you think all the information about branches need to go into the CONTRIBUTING.md? I see the CONTRIBUTING.md as a way for external developers to get started (thus, the mentioning of the CLA). Do external contributors need to know about branching? Seems like they should always be making contributions from **develop**. Maybe have a separate branching.md file in the docs/ folder?
1. Could you change all the numbering to 1. 1. 1. I know you have a preference but let's keep it consistent in the repo.

**Comment by guw** [comment]
I was thinking about adding a separate doc and I find it simplier/easier to have one (well structures) contribution guide for now. Have it in on place makes it easier to find and also improves visibility/transparency about out process. Some of the information is valuable to both groups (external contributors as well as committers).

I'll work on the formatting changes.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
+  - Instead we aim for faster cycles and a generally stable `develop` branch.
+
+### Merging `develop` into `master`
+* When a development cycle finishes, the content of the `develop` branch will become the `master` branch
+```
+$ git checkout master
+$ git reset --hard develop
+$
+$ # Using a custom commit message for the merge below
+$ git merge -m 'Merge -s our (where _ours_ is develop) releasing stream x.y.z.' -s ours origin/master
```
Comment (ruthemmanuelle) [inline CONTRIBUTING.md]: This commit message is confusing to me. Can it be rephrased?

**Comment by vazexqi** [comment]
@guw - LGTM 👍

**Comment by jamesward** [comment]
I don't totally get from this why the `develop` branch exists.  Most OSS projects use `master` as the "good to go" branch and then feature / bug fix branches.

**Comment by vazexqi** [comment]
@jamesward This is based on gitflow (https://danielkummer.github.io/git-flow-cheatsheet/). The SFDX projects use this model for releasing.

**Comment by jamesward** [comment]
In practice I haven't really seen OSS projects using the `develop` branch and some consider it redundant: http://endoflineblog.com/gitflow-considered-harmful

Either way, I think the contrib doc should provide more details to the difference between `develop` and `master` since this is pretty atypical of OSS.

**Pull request merged**